libs/data-types/src/lib/utilities/bootstrap-dynamic-types/resolve-member-schemas-for-union.ts

```diff
@@ -0,0 +1,76 @@
+import { UnionMemberMetadata } from '../../decorators';
+import { ClassSchema, isSimpleCoscradPropertyTypeDefinition } from '../../types';
+import getCoscradDataSchema from '../getCoscradDataSchema';
+import { buildUnionTypesMap } from './bootstrap-dynamic-types';
+import { leveragesUniontype } from './leverages-union-type';
+
+export type UnionMemberSchemaDefinition = Pick<UnionMemberMetadata, 'discriminantValue'> & {
+    schema: ClassSchema;
+};
+
+export const resolveMemberSchemasForUnion = (
+    allCtorCandidates: unknown[],
+    unionName: string
+): UnionMemberSchemaDefinition[] => {
+    const unionMap = buildUnionTypesMap(allCtorCandidates);
+
+    const membersMap = unionMap.get(unionName);
+
+    if (!membersMap) {
+        return [];
+    }
+
+    return Array.from(membersMap.keys()).map((discriminantValue) => {
+        const ctor = membersMap.get(discriminantValue);
+
+        const rawSchema = getCoscradDataSchema(ctor);
+
+        const updatedSchema = Object.entries(rawSchema).reduce(
+            (acc, [propertyKey, propertyTypeDefinition]) => {
+                if (
+                    isSimpleCoscradPropertyTypeDefinition(propertyTypeDefinition) ||
+                    !['NESTED_TYPE', 'UNION_TYPE'].includes(propertyTypeDefinition.complexDataType)
+                )
+                    return {
+                        ...acc,
+                        [propertyKey]: propertyTypeDefinition,
+                    };
+
+                const { complexDataType } = propertyTypeDefinition;
+
+                if (complexDataType === 'NESTED_TYPE') {
+                    const { schema: nestedSchema } = propertyTypeDefinition;
+
+                    // @ts-expect-error fix types
+                    if (!leveragesUniontype(nestedSchema))
+                        return {
+                            ...acc,
+                            [propertyKey]: propertyTypeDefinition,
+                        };
+
+                    throw new Error(
+                        `not implemented: cannot resolve union type: ${unionName} as one of its members leverages another union`
```

Comment:
Should we add a test that hits this and expects the exception? That would be a good way to document the current lack of support for unions that leverage further unions.